src/pages/Facility/settings/billing/discount/discount-components/DiscountMonetaryComponentForm.tsx

```diff
@@ -162,19 +222,21 @@ export function DiscountMonetaryComponentForm({
                       <FormControl>
                         <div className="relative">
                           <Input
-                            type="number"
-                            min="0"
-                            max="100"
-                            step="0.01"
+                            type=""
                             {...field}
-                            onChange={(e) =>
-                              field.onChange(
-                                e.target.value
-                                  ? parseFloat(e.target.value)
-                                  : null,
-                              )
+                            value={
+                              typeof field.value === "number" ||
+                              typeof field.value === "string"
+                                ? field.value
+                                : ""
                             }
-                            value={field.value === null ? "" : field.value}
+                            onChange={(e) => {
+                              const numbersOnly = e.target.value.replace(
```

Comment:
same here, shouldn't need all this, let's simplify. Let zod handle validation, let's not replace invalid characters.